File: src/trees/SameTree.java
package trees;

public class SameTree {

    private class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
        TreeNode() {}
        TreeNode(int val) { this.val = val; }
        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if(p == null || q == null){
            return p == q;
        }
        Boolean left = isSameTree(p.left, q.left);
        Boolean right = isSameTree(p.right,q.right);
        return left && right && p.val == q.val;
    }
}

File: src/Trees/SameTree.kt
package Trees

class SameTree {
    data class TreeNode(val `val`: Int, val left: TreeNode?, val right: TreeNode?)
    fun isSameTree(p: TreeNode?, q: TreeNode?): Boolean {
        if (p == null || q == null){
            return p == q
        }
        return isSameTree(p.left, q.left) && isSameTree(p.right, q.right) && p.left?.`val` == q.left?.`val`
    }
}